### DISCUSSION TITLE
[5] AJAX loading multiple templates

### DISCUSSION THREAD

**Discussion by Coysh** [solspace/craft-freeform#1088] [Questions & Answers] [answered]
**Describe the bug or issue you're experiencing**
When a multi-page form is used with AJAX, the template can be loaded multiple times.

![CleanShot 2024-02-22 at 12 06 43@2x](https://github.com/solspace/craft-freeform/assets/5764356/d4fa447a-0ee0-4c83-bb2d-2cd4c2291964)

Template code (bottom)
```
{# Display form field rows and columns #}
{{ rowMacro.render(form.rows, form) }}

{# Render the closing form tag #}
{{ form.renderClosingTag }}

<div class="flex flex-wrap -mx-3{{ form.customAttributes.rowClass }}">
	<div class="main-column-class w-full px-3 mb-6 md:w-full lg:w-full xl:w-full text-sm">
		{{ siteWide.dataProtectionStatement }}
	</div>
</div>
```

**Steps to reproduce**

1. Create AJAX form multipaged
2. Go to next page
3. Data Protection statement is displayed twice

**Expected behavior**
The data protection statement should be seen once

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 4.7.3
- Freeform Version: 5.0.12
- Freeform Edition:Pro
- Fresh Install or Upgrade:

**Additional context**

**Comment by gustavs-gutmanis** [comment, marked as answer]
Hi @Coysh,

You're rendering the `dataProtectionStatement` outside of the `form.renderClosingTag`.  
Freeform's AJAX loads a re-rendered form HTML and scripts after a successful submit and replaces the current `<form>...</form>` with those. Since your data protection block is rendered outside, the initial render remains unreplaced, and the replaced HTML contains another one.

**Comment by Coysh** [comment]
Thanks @gustavs-gutmanis - that is helpful
